dune/grid/cpgrid/processEclipseFormat.cpp

```diff
@@ -99,12 +103,52 @@ namespace cpgrid
         std::vector<double> coordData;
         ecl_grid.exportCOORD(coordData);
 
-        std::vector<double> zcornData;
-        ecl_grid.exportZCORN(zcornData);
-
         std::vector<int> actnumData;
         ecl_grid.exportACTNUM(actnumData);
 
+        std::vector<double> zcornData;
+        {
+            ecl_grid.exportZCORN(zcornData);
+
+            auto repair = ::Opm::UgGridHelpers::RepairZCORN {
+                std::move(zcornData), actnumData,
+                std::vector<int>{ ecl_grid.getNX() ,
+                                  ecl_grid.getNY() ,
+                                  ecl_grid.getNZ() }
+            };
+
+            zcornData = repair.destructivelyGrabSanitizedValues();
+
+            if (repair.switchedToDepth()) {
+                std::cout << "ZCORN Values Switched from Elevation to "
```

Comment:
These should eventually be replaced with OpmLog::info() or OpmLog::warning(), but then the upscaling programs must also be changed to set up logging (preferably just using the OpmLog::setupSimpleDefaultLogging() function) so that can wait. We do risk getting messy output from the simulators though, if triggered.